### PR TITLE
[Test] Rework PR Gate

### PR DESCRIPTION
**Pull request by riedgar-ms**
Trying to improve how the GitHub workflows are built. Inspired by #758 by @ryanpeach

### CODE REVIEW

**Comment by ryanpeach** [comment]
Very clean! And I've never done it that way so glad to learn.

**Comment by Harsha-Nori** [comment]
I think this looks reasonable to me. @nopdive might have some thoughts too

File: .github/workflows/workflow-pr-gate.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    needs:
+      - unit-tests-linux-python-other
+      - unit-tests-gpu-python-latest
+    name: End Stage 2
+    runs-on: ubuntu-latest
+    steps:
+    - run: echo "End of Stage 1"
+
+# Third Stage ==============================================================
+# Windows and MacOS, plus other GPU Linux tests
```
Comment (Harsha-Nori) [inline .github/workflows/workflow-pr-gate.yml:122]: Appreciate the cascaded staging :), and the structure seems reasonable!

**Comment by riedgar-ms** [comment]
Have merged for now, but am willing to make further changes in future. The cascading of stages is a bit involved; I'd happily simplify it, if we were testing fewer Python versions

**Pull request merged**